iree/turbine/kernel/wave/index_sequence_analysis.py

```diff
@@ -80,25 +81,42 @@ def has_strided_access(node: fx.Node) -> bool:
             for dim in custom.index
         }
 
-        max_stride = int(max(simplified_index[dim].stride for dim in simplified_index))
         shape = get_vector_shape(hw_constraint, custom.register_type.symbolic_shape)
         elements_per_thread = subs_idxc(custom.elements_per_thread)
+        max_stride_dim, max_stride = max(
+            [(dim, seq.stride) for dim, seq in simplified_index.items()],
+            key=lambda item: item[1],
+        )
         with custom.graph.inserting_before(operator):
             for i in range(elements_per_thread):
+                # Non-contiguous access patterns can have varying offsets. We
+                # handle that here.
+                gpr_offset = [
+                    expr
+                    for expr in simplified_index[max_stride_dim].start.args
+                    if expr.has(GPR_NUM)
```

Comment:
just curious wouldn't the indexSequence.start.arg be singular symbols? so wouldn't `expr == GPR_NUM` be more apt here?

Reply:
or would expr be:
```
[ (8 * floor(GPR_NUM / 4) % 32), 4 * floor(lane / 32), (GPR_NUM % 4)] ?
 ```

Reply:
The idea is to separate the offset into a gpr_offset and lane_offset. We compute the gpr_offset here and add it to the lane offset later.